lib/requester/request-wrapper.js

```diff
@@ -78,16 +79,18 @@ var _ = require('lodash'),
     setProxy = function(options, cb) {
         var url = _.get(options, 'request.url');
 
-        if (url && options.proxyManager && options.proxyManager.getProxyConfiguration) {
+        if (url && options.proxyList) {
+            if(!_.isFunction(options.proxyList.isProxyConfigList)) {
+              options.proxyList = new ProxyConfigList({}, options.proxyList);
```

Comment:
@shamasis `options.proxyList` can be an instance of `ProxyList` or a POJO. Since the list is created in the browser process, it cannot be passed as-is into the Node process. Which is why this is correct IMO.